distributed/utils_test.py

```diff
@@ -1761,3 +1761,54 @@ async def connect(self, *args, **kwargs):
         return LockedComm(
             comm, self.read_event, self.read_queue, self.write_event, self.write_queue
         )
+
+
+def assert_worker_story(story: list[tuple], expect: list[tuple]) -> None:
+    """Test the output of ``Worker.story``
+
+    Parameters
+    ==========
+    story: list[tuple]
+        Output of Worker.story
+    expect: list[tuple]
+        Expected events. Each expected event must contain exactly 2 less fields than the
+        story (the last two fields are always the stimulus_id and the timestamp).
+
+    story may contain more events than expect. Extra events are ignored.
```

Comment:
I'm wondering if this is too relaxed. Consider

```python
[
    "ensure-exists",
    ("released", "waiting"),
    ("waiting", "executing"),
    ("executing", "memory"),
]
```

is quite different than

```python
[
    "ensure-exists",
    ("released", "waiting"),
    ("waiting", "executing"),
    ("executing", "cancelled"),
    ("cancelled", "released"),),
    ("released", "waiting"),
   ...
    ("executing", "memory"),
]
```

Reply:
I would assume we need both the relaxed and strict version occasionally?

Reply:
Added strict mode, please review